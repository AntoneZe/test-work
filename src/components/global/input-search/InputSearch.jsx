import React, { useState } from "react";
import PropTypes from "prop-types";

import "./input-search.less";

const useSearch = (callback) => {
  const searchCallback = (e, value) => {
    e.preventDefault();
    callback(value);
  };

  return [searchCallback];
};

const InputSearch = ({ callback }) => {
  const [value, setValue] = useState("");
  const [searchCallback] = useSearch(callback);

  return (
    <div className="custom-input-search">
      <form>
        <div>
          <div className="custom-input-search__form-wrapper">
            <input
              type="search"
              id="mySearch"
              name="q"
              value={value}
              onChange={(e) => setValue(e.target.value)}
              placeholder="Search the site..."
            />
          </div>
          <button onClick={(e) => searchCallback(e, value)}>
            <img src="public/img/icons-search.svg" alt="альтернативный текст" />
          </button>
        </div>
      </form>
    </div>
  );
};

InputSearch.propTypes = {
  callback: PropTypes.func.isRequired,
};

export default InputSearch;
