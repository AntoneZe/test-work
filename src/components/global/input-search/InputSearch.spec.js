import React from "react";

import InputSearch from "./InputSearch";

describe("InputSearch component", () => {
  it("should render InputSearch component", () => {
    const component = render(<InputSearch callback={() => {}} />);
    expect(component).toMatchSnapshot();
  });
});

describe("InputSearch search", () => {
  it("should reset search", () => {
    const setValue = jest.fn();
    const event = {
      preventDefault() {},
      target: {
        value: "",
      },
    };

    const component = mount(<InputSearch callback={setValue} />);

    component.find("input").simulate("change", event);
    expect(setValue.mock.calls.length).toEqual(1);
  });

  it("should do search", () => {
    const setValue = jest.fn();
    const handleClick = jest.spyOn(React, "useState");
    handleClick.mockImplementation((value) => [value, setValue]);

    const event = {
      preventDefault() {},
      target: {
        value: "test",
      },
    };

    const component = mount(<InputSearch callback={() => {}} />);

    component.find("input").simulate("change", event);
    expect(component.find("input").prop("value")).toEqual("test");
  });
});

describe("InputSearch search button", () => {
  it("InputSearch search button do once", () => {
    const callback = jest.fn();
    const component = mount(<InputSearch callback={callback} />);

    const event = {
      preventDefault() {},
      target: {
        value: "",
      },
    };

    component.find("button").simulate("click", event);
    expect(callback.mock.calls.length).toEqual(1);
  });
});
