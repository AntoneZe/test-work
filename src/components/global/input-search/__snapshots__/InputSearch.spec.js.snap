// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSearch component should render InputSearch component 1`] = `
<div
  class="custom-input-search"
>
  <form>
    <div>
      <div
        class="custom-input-search__form-wrapper"
      >
        <input
          id="mySearch"
          name="q"
          placeholder="Введите Имя..."
          type="search"
          value=""
        />
      </div>
      <button>
        <img
          alt="альтернативный текст"
          src="public/img/icons-search.svg"
        />
      </button>
    </div>
  </form>
</div>
`;
